model/src/main/java/uk/gov/pay/commons/api/validation/ExternalJsonMetadataValidator.java

```diff
@@ -1,93 +1,90 @@
 package uk.gov.pay.commons.api.validation;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import fj.data.List;
+import fj.data.Stream;
 
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
-import java.util.HashSet;
-import java.util.Set;
+import java.util.function.BiFunction;
+
+import static fj.data.List.arrayList;
+import static java.lang.String.format;
 
 public class ExternalJsonMetadataValidator implements ConstraintValidator<ValidExternalJsonMetadata, JsonNode> {
 
     private static final int MAXIMUM_KEYS = 10;
     private static final int MAXIMUM_VALUE_LENGTH = 50;
     private static final int MAXIMUM_KEY_LENGTH = 30;
 
+    private static final BiFunction<JsonNode, List<String>, List<String>> VALIDATE_KEY_VALUE_PAIRS = (jsonNode, errors) -> {
+        if (jsonNode.isArray()) return arrayList("must be an object of JSON key-value pairs").append(errors);
```

Comment:
This might be better named as `VALIDATE_IS_OBJECT` and make the check `!jsonNode.isObject()` since whilst using it with the existing check I was able to pass a `TextNode` in which is basically a String and it didn't complain at all.